tests/FunctionsFrameworkTest.php

```diff
@@ -56,6 +59,20 @@ public function testRegisterAndRetrieveCloudEventFunction(): void
         );
     }
 
+    public function testRegisterAndRetrieveTypedFunction(): void
+    {
+        $fn = function (IntValue $event) {
+            return $event;
+        };
+
+        FunctionsFramework::typed('testFn', $fn);
+
+        $this->assertEquals(
+            $fn,
+            FunctionsFrameworkTesting::getRegisteredFunction('testFn')
+        );
+    }
```

Comment:
I'd like to see more testing here. What happens when the expected signature doesn't match?

Reply:
Hmm, the existing pattern is using a test helper `FunctionsFrameworkTesting` which applies reflection to find the loaded function. Do you think it's worth expanding that reflection to provide a utility to also check the types of registered functions for this and the other declaratively registered function types?

Reply:
I guess what I mean is I'd like to see a test for what happens if a function is registered which has a bad signature

Reply:
Done, added a test that checks that an error is thrown and that the function isn't registered.